worker/raft/queue/blockingqueue_test.go

```diff
@@ -245,16 +177,55 @@ func (s *BlockingOpQueueSuite) TestMultipleEnqueues(c *gc.C) {
 			defer wg.Done()
 
 			err := queue.Enqueue(Operation{
-				Command:  cmd(i),
-				Deadline: clock.WallClock.Now().Add(time.Second),
+				Command:  opName(i),
+				Deadline: now.Add(time.Second),
 			})
 			c.Assert(err, jc.ErrorIsNil)
 		}(i)
 	}
 	wg.Wait()
 
-	c.Assert(results, jc.SameContents, []string{
+	var received []string
+	for result := range results {
+		received = append(received, string(result))
+	}
+	c.Assert(len(received), gc.Equals, 10)
+	c.Assert(received, jc.SameContents, []string{
 		"abc-0", "abc-1", "abc-2", "abc-3", "abc-4",
 		"abc-5", "abc-6", "abc-7", "abc-8", "abc-9",
 	})
 }
+
+func opName(i int) []byte {
+	return []byte(fmt.Sprintf("abc-%d", i))
+}
+
+func consumeN(queue *BlockingOpQueue, n int) <-chan []byte {
+	return consumeNUntilErr(queue, n, nil)
+}
+
+func consumeNUntilErr(queue *BlockingOpQueue, n int, err error) <-chan []byte {
+	results := make(chan []byte, n)
+
+	go func() {
+		defer close(results)
+
+		var count int
+		for op := range queue.Queue() {
+			results <- op.Command
```

Comment:
You might want to wrap this in a select where you also add the LongWait check to prevent broken tests from blocking forever. Thought that means that you need to thread `gc.C` through